### PR TITLE
Add new classes and refactor the code

### PR DESCRIPTION
**Pull request by wolfrednicolas**
- add and remove dependencies.
- modify the model class reference in the config file.
- rename the factory and add the the definition method.
- add the migration for the table and rename the file.
- rename the command and change the signature.
- create the Model's folder move the class, rename it and add the methods.
- rename the facade and change the return value for the method getFacadeAccessor.
- rename the service provider, change the method configurePackage and create the method packageBooted.

### CODE REVIEW

File: config/flexdays.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
```
Comment (wolfrednicolas) [inline config/flexdays.php:1]: I want to confirm if the name of the file is ok? @drewroberts

File: src/Commands/FlexDayCommand.php [hunk 2]
```diff
@@ -4,9 +4,9 @@
```
Comment (wolfrednicolas) [inline src/Commands/FlexDayCommand.php:4]: I want to confirm if the signature name is ok? @drewroberts

File: src/FlexDayFacade.php [hunk 3]
```diff
@@ -5,12 +5,12 @@
...
- * @see \Tipoff\FlexScheduling\FlexScheduling
+ * @see \Tipoff\FlexScheduling\FlexDayFacade
  */
-class FlexSchedulingFacade extends Facade
+class FlexDayFacade extends Facade
 {
     protected static function getFacadeAccessor()
     {
-        return 'flex-scheduling';
+        return 'flexdays';
```
Comment (wolfrednicolas) [inline src/FlexDayFacade.php:14]: I want to confirm if the return value is ok? @drewroberts

File: src/Commands/FlexDayCommand.php (hunk 2 above)
Comment (drewroberts) [reply src/Commands/FlexDayCommand.php:4]: yes, that's good for me.

File: src/FlexDayFacade.php (hunk 3 above)
Comment (drewroberts) [reply src/FlexDayFacade.php:14]: Yes, that works for me.

File: config/flexdays.php (hunk 1 above)
Comment (drewroberts) [reply config/flexdays.php:1]: Let's make the name of the config file consistent with the package name. So, let's go with `config/flex-scheduling.php`

**Comment by wolfrednicolas** [comment]
@drewroberts I removed the label `help-wanted` and I think is ready to merge, however I notice one of the test is failing, I checked other packages and contains the same exactly code :octocat:

**Review by drewroberts — APPROVED** [review]
Thank you, Wolfred. 🌴

**Pull request merged**